use-package.el

```diff
@@ -419,23 +419,23 @@ manually updated package."
   (let ((archive-symbol (if (symbolp archive) archive (intern archive)))
         (archive-name   (if (stringp archive) archive (symbol-name archive))))
     (if (use-package--archive-exists-p archive-symbol)
-        (push (cons package archive-name) package-pinned-packages)
+        (add-to-list 'package-pinned-packages (cons package archive-name) t)
```

Comment:
Well, actually, I'm not sure if this change is safe.
`push` is a macro, where `add-to-list` is a function.
`add-to-list` docs say:

>This is handy to add some elements to configuration variables,
but please do not abuse it in Elisp code, where you are usually
better off using `push' or `cl-pushnew'.

What do you think?

Maybe replace it with `member` check + `push`?